mbed_greentea/mbed_test_api.py

```diff
@@ -123,7 +123,7 @@ def run_host_test(image_path,
                   global_resource_mgr=None,
                   num_sync_packtes=None,
                   tags=None,
-                  run_app=None):
+                  run_app=None): # pragma: no cover
```

Comment:
Could you explain why we don't want this function to count towards our coverage?

Reply:
I may have forgotten to remove that when I was cleaning up.

However, as with main_cli and the other top-level commands, it doesn't seem like a reasonable function to cover with unit testing; It takes a lot of arguments and aggregates a number of actually testable function calls.

Whilst it would be fairly trivial to cover all the lines in testing, that might be slightly disingenuous because covering all the branches/behaviour would be a lot of work and far from simple.

Reply:
Would it also be slightly disingenuous to remove coverage from them?

Reply:
In a way. But, I'd argue that it *shouldn't* be tested, rather than *hasn't* been tested, which excluding it would indicate clearer.

Reply:
> I'd argue that it shouldn't be tested

I would not argue that.

> than hasn't been tested

This is definitely true.

I'd rather see these lines as red in a coverage report, rather than gray. That's just my opinion.